python_modules/dagster/dagster/core/log_manager.py

```diff
@@ -149,13 +149,24 @@ def construct_log_string(
 
 
 class DagsterLogManager(logging.Logger):
-    def __init__(self, logging_metadata: DagsterLoggingMetadata, loggers: List[logging.Logger]):
+    def __init__(
+        self,
+        logging_metadata: DagsterLoggingMetadata,
+        loggers: List[logging.Logger],
+        handlers: Optional[List[logging.Handler]] = None,
+    ):
         self._logging_metadata = check.inst_param(
             logging_metadata, "logging_metadata", DagsterLoggingMetadata
         )
         self._loggers = check.list_param(loggers, "loggers", of_type=logging.Logger)
+
         super().__init__(name="dagster", level=logging.DEBUG)
 
+        check.opt_nullable_list_param(handlers, "handlers", of_type=logging.Handler)
```

Comment:
this can be simplified by using check.opt_list_param(), which will return an empty list (instead of None) if the input is None. that way you don't need to do a None check before the loop